datafusion/sqllogictest/src/filters.rs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use datafusion::sql::parser::{DFParserBuilder, Statement};
+use sqllogictest::{AsyncDB, Record};
+use sqlparser::ast::{SetExpr, Statement as SqlStatement};
+use sqlparser::dialect::dialect_from_str;
+use std::path::Path;
+use std::str::FromStr;
+
+/// Filter specification that determines whether a certain sqllogictest record in
+/// a certain file should be filtered. In order for a [`Filter`] to match a test case:
+///
+/// - The test must belong to a file whose absolute path contains the `file_substring` substring.
+/// - If a `line_number` is specified, the test must be declared in that same line number.
+///
+/// If a [`Filter`] matches a specific test case, then the record is executed, if there's
+/// no match, the record is skipped.
+#[derive(Debug, Clone)]
```

Comment:
I think it would also help to add an example here of what a filter looks like

For example

```suggestion
///
/// Filters can be parsed from strings of the form `<file_name>:line_number`. For example:
/// `foo.slt:100` matches any test on line 100 from any file whose name contains `foo.slt`
#[derive(Debug, Clone)]
```